### DISCUSSION TITLE
Adding a bio inspired algorithm for NAS

### DISCUSSION THREAD

**Discussion by abinezer** [microsoft/nni#4348] [Ideas]
I have written a paper which is currently under review. I suggested an approach wherein a bio-inspired algorithm could be used to suggest the number of hidden layers and number of neurons in each hidden layer given a dataset, for a neural network. I was wondering how i could incorporate my algorithm into the nni repo. Would appreciate any info and insights..!

**Comment by QuanluZhang** [comment]
@abinezer thanks for your interest in nni. We always welcome external contributions to nni. Could share the paper or describe the main logic of your algorithm, so that we can give our feedback on possible integration plans.
